plugins/source/aws/resources/services/cloudwatchlogs/metric_filters_fetch.go

```diff
@@ -31,12 +31,13 @@ func fetchCloudwatchlogsMetricFilters(ctx context.Context, meta schema.ClientMet
 
 func resolveMetricFilterArn(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
 	cl := meta.(*client.Client)
+	r := resource.Item.(types.MetricFilter)
 	a := arn.ARN{
 		Partition: cl.Partition,
 		Service:   "logs",
 		Region:    cl.Region,
 		AccountID: cl.AccountID,
-		Resource:  "metric_filter/" + aws.ToString(resource.Item.(types.MetricFilter).FilterName),
+		Resource:  "log-group:" + aws.ToString(r.LogGroupName) + ":metric_filter:" + aws.ToString(r.FilterName),
```

Comment:
inspired from https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazoncloudwatchlogs.html#amazoncloudwatchlogs-resources-for-iam-policies

The other "clean fix" is to remove this field and add a composite PK for account+region+log group+metric name.

Reply:
I think there's no such field as metric filter ARN, so I'm for removing this field and doing composite PK

Reply:
Done in [`04e2744` (#8061)](https://github.com/cloudquery/cloudquery/pull/8061/commits/04e2744c90f228341eb50fa49da65c7f85c818c8)